### ISSUE TITLE
Accessibility Feedback - Midpoint Review - BAH/Education Benefits VA Form 22-5490, 22-0994, 22-1990E

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#26936]
## VFS acceptance criteria
- [x] Let Platform know when the **Must** feedback has been incorporated 
- [x] Leave any comments for feedback you decide _not_ to take
- [x] VFS team closes the ticket

### Must

* Screen 1 of 14: Learn more link under Spouse or surviving spouse radio button must have an accessible label so screen reader users have context.
* Screen 2 of 14: The message about applicants under the age of 18 must announce itself to screen readers without user input. This means the container might need an `aria-live` attribute, or another way to announce this as a message in the natural speaking flow.
* Screen 8 of 14: The Fry Scholarship must announce the information to screen readers without user input. This might be better served with an `aria-describedby` that is added to the input when users check the radio button.
* Screens 11, 12, 13: Same comment about needing an `aria-describedby` to announce the month, day, year set of inputs depending on the checked radio button. This may become "noisy", but the attribute allows screen readers to announce the current radio button, pause, then read the describedby text like "Sponsor's date of..."

### Should

* Screen 14: Submit application button should have an `aria-describedby` attribute that references the "By submitting this form you certify that..." paragraph and bullet points. This way screen readers have a second chance to hear what they are attesting to in case they skipped to the submit button and bypassed the text information.
* I don't have a social security number should be spaced farther away from the input. This could cause an issue for mobile device users (tap gestures).
* Other benefit should hide the text input until users check the box. A disabled input goes against current patterns and might be confusing to users how to make it active.

### Consider

* Updating the TTY phone number to TTY: 711 in the footer. Danielle on sitewide content will have the final information, but I believe we are moving to a single 711 service in favor of other 800 numbers.

### Questions

* Screen 2 of 14: Does the message "Applicants under the age of 18..." appear when the year field loses focus?

### Consider

* Updating the TTY phone number to TTY: 711 in the footer. Danielle on sitewide content will have the final information, but I believe we are moving to a single 711 service in favor of other 800 numbers.

### Should

* Submit application button should have an `aria-describedby` attribute that references the "By submitting this form you certify that..." paragraph and bullet points. This way screen readers have a second chance to hear what they are attesting to in case they skipped to the submit button and bypassed the text information.

### Consider

* Updating the TTY phone number to TTY: 711 in the footer. Danielle on sitewide content will have the final information, but I believe we are moving to a single 711 service in favor of other 800 numbers.

## Feedback
- **Must** = the feedback must be applied
- **Should** = the feedback is best practice
- **Consider** = suggestions/enhancements

**Comment by amyknox** [comment]
Thanks for the input Trevor.  Question for you on "aria-describedby":  Curious if that label is added to the question content or is it on the input control? 

👀 @1Copenut

**Comment by 1Copenut** [comment]
Great question @amyknox ! The `aria-describedby="UNIQUE_ID"` attribute is added to the input that toggles the new questions or controls. It should point to a unique ID on the new questions/controls container so the input will read out the new elements.

**Comment by amyknox** [comment]
Got it.  Thanks @1Copenut !!

**Comment by dneel-bah** [comment]
@1Copenut To be clear then, this request:
- Screens 11, 12, 13: Same comment about needing an aria-describedby to announce the month, day, year set of inputs depending on the checked radio button. This may become "noisy", but the attribute allows screen readers to announce the current radio button, pause, then read the describedby text like "Sponsor's date of..."

Would actually reference the unique ID of the label for the date fields as opposed to the date fields themselves, so it would look something like (forgive my terrible markup):
<input type="radio" aria-describedby="Sponsor's date of death label UNIQUEID">
Died while serving on active duty or duty other than active duty</input>

**Comment by dneel-bah** [comment]
@1Copenut Also, to answer this question:

Screen 2 of 14: Does the message "Applicants under the age of 18..." appear when the year field loses focus?

Yes, but more specifically it will display once the last piece of the date is populated that calculates the DOB to be < 18.  So for example if the applicant filled in the month, tabbed over the day, filled in the year, selected a day it would display after the full date was available to verify the age is < 18.

**Comment by dneel-bah** [comment]
**5490**
#26436 - Actively working in the current/next sprint
   - All musts included
   - Should's 1 & 3 included

Not going to take:
   - Should 2: This is a part of a platform definition, and similar behavior is contained in other forms.  Added to backlog #26996
   - Consider 1: Our mockup templates may be out of date as the "Need Help" footer has the TTY: 711.

**1990E**
Not going to take:
   - Consider 1: Our mockup templates may be out of date as the "Need Help" footer has the TTY: 711.



**0994**
#26995
- Should 1 issue created and added to backlog

Not going to take:
   - Consider 1: Our mockup templates may be out of date as the "Need Help" footer has the TTY: 711.

**Comment by amyknox** [comment]
Double checking the templates, the Sketch template that was used appears to be the most recent (date updated: May 25).  Since the footer is a shared component, we'll be ok regardless of what displays in the prototype footer.

**Comment by 1Copenut** [comment]
@dneel-bah I'm hoping we can announce the new label plus inputs on screens 11-13 when a radio button is selected. Something like the mockup below. I'd also like to ask if we can make the label "Sponsor's date of death" be more specific for the while on duty or service-connected disability labels?

![Screen Shot 2021-07-09 at 2 39 04 PM](https://user-images.githubusercontent.com/934879/125128681-f8a8f900-e0c3-11eb-92f8-bdc1cd642956.png)

**Comment by 1Copenut** [comment]
@elder-joseph I filed issue https://github.com/department-of-veterans-affairs/va.gov-team/issues/28404 with the Design System team so you can provide an improved experience here.

**Comment by dneel-bah** [comment]
## 5490

### Must

* Screen 1 of 14: Learn more link under Spouse or surviving spouse radio button must have an accessible label so screen reader users have context. 

    > Addressed in #26436.

* Screen 2 of 14: The message about applicants under the age of 18 must announce itself to screen readers without user input. This means the container might need an `aria-live` attribute, or another way to announce this as a message in the natural speaking flow.

    > Addressed in #26436.

* Screen 8 of 14: The Fry Scholarship must announce the information to screen readers without user input. This might be better served with an `aria-describedby` that is added to the input when users check the radio button. 

    > Addressed in platform issue #28404.

* Screens 11, 12, 13: Same comment about needing an `aria-describedby` to announce the month, day, year set of inputs depending on the checked radio button. This may become "noisy", but the attribute allows screen readers to announce the current radio button, pause, then read the describedby text like "Sponsor's date of..."

    > Addressed in #26436.

### Should

* Screen 14: Submit application button should have an `aria-describedby` attribute that references the "By submitting this form you certify that..." paragraph and bullet points. This way screen readers have a second chance to hear what they are attesting to in case they skipped to the submit button and bypassed the text information. 

    > Recommend this be handled by the platform team.  See [comment](https://github.com/department-of-veterans-affairs/va.gov-team/issues/26936#issuecomment-879167946).

* I don't have a social security number should be spaced farther away from the input. This could cause an issue for mobile device users (tap gestures).  

    > Created story #26996; this is used in many forms and would need to be checked in each one after updating.

* Other benefit should hide the text input until users check the box. A disabled input goes against current patterns and might be confusing to users how to make it active.

    > Addressed in #26436.

### Consider

* Updating the TTY phone number to TTY: 711 in the footer. Danielle on site wide content will have the final information, but I believe we are moving to a single 711 service in favor of other 800 numbers.

    > This appears to have been resolved by the platform team.

**Comment by dneel-bah** [comment]
## 1990e

### Consider

* Updating the TTY phone number to TTY: 711 in the footer. Danielle on sitewide content will have the final information, but I believe we are moving to a single 711 service in favor of other 800 numbers.

    > This appears to have been resolved by the platform team.

**Comment by dneel-bah** [comment]
## 0994

### Should

* Submit application button should have an `aria-describedby` attribute that references the "By submitting this form you certify that..." paragraph and bullet points. This way screen readers have a second chance to hear what they are attesting to in case they skipped to the submit button and bypassed the text information.

    > Created story #26995 and added to backlog.

### Consider

* Updating the TTY phone number to TTY: 711 in the footer. Danielle on sitewide content will have the final information, but I believe we are moving to a single 711 service in favor of other 800 numbers.

    > This appears to have been resolved by the platform team.

**Comment by dneel-bah** [comment]
All musts have been addressed and tickets have been created for the remaining items.  Closing this issue.
